test/dynamo/test_error_messages.py

```diff
@@ -1182,6 +1183,25 @@ def f3(x):
 """,
         )
 
+    @make_logging_test(dynamo=logging.DEBUG)
+    def test_lru_cache_warning_logs_user_stack_trace(self, records):
+        from functools import lru_cache
+
+        @lru_cache
+        def foo(x):
+            return x + 1
+
+        torch.compile(foo, backend="eager")(torch.randn(4))
+
+        self.assertTrue(
+            any(
+                "call to a lru_cache` wrapped function from user code at:"
+                in record.getMessage()
+                for record in records
+            ),
+            "No lru_cache warning was logged",
+        )
```

Comment:
Please also add a `assertExpectedInlineMunged` check so that we can see what the user code traceback looks like.